src/compile/axis/properties.ts

```diff
@@ -167,58 +169,111 @@ export function getLabelAngle(
   }
 }
 
+export function normalizeAngleExpr(angle: SignalRef) {
+  return `(((${angle.signal} % 360) + 360) % 360)`;
+}
+
 export function defaultLabelBaseline(
-  angle: number,
-  axisOrient: AxisOrient,
+  angle: number | SignalRef,
+  orient: AxisOrient | SignalRef,
   channel?: 'x' | 'y',
   alwaysIncludeMiddle?: boolean
 ) {
-  channel = channel || (axisOrient === 'top' || axisOrient === 'bottom' ? 'x' : 'y');
+  channel = channel || (orient === 'top' || orient === 'bottom' ? 'x' : 'y');
 
   if (angle !== undefined) {
     if (channel === 'x') {
-      return (45 < angle && angle < 135) || (225 < angle && angle < 315)
-        ? 'middle'
-        : (angle <= 45 || 315 <= angle) === (axisOrient === 'top')
-        ? 'bottom'
-        : 'top';
+      if (isSignalRef(angle)) {
+        const a = normalizeAngleExpr(angle);
+        const orientIsTop = isSignalRef(orient) ? `(${orient.signal} === "top")` : orient === 'top';
```

Comment:
I believe we need to parenthesize `${orient.signal}` here as well.

Reply:
Can you come up with a case that will break this?
Otherwise, I'd rather have the output readable.

Reply:
Let's deal with this in https://github.com/vega/vega-lite/issues/6374